manim/mobject/types/vectorized_mobject.py

```diff
@@ -1017,6 +1041,50 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
 
             current_length += length
 
+    def proportion_from_point(
+        self,
+        point: typing.Iterable[typing.Union[float, int]],
+    ) -> float:
+        """Returns the proportion along the path of the :class:`VMobject`
+        a particular given point is at.
+
+        Parameters
+        ----------
+        point
+            The Cartesian coordinates of the point which may or may not lie on the :class:`VMobject`
+
+        Returns
+        -------
+        float
+            The proportion along the path of the :class:`VMobject`.
+
+        Raises
+        ------
+        :exc:`ValueError`
+            If ``point`` does not lie on the curve.
+        :exc:`Exception`
+            If the :class:`VMobject` has no points.
+        """
+        self.throw_error_if_no_points()
+
+        num_curves = self.get_num_curves()
+        total_length = self.get_arc_length()
+        target_length = 0
+        for n in range(num_curves):
+            control_points = self.get_nth_curve_points(n)
+            length = self.get_nth_curve_length(n)
+            if point_lies_on_bezier(point, control_points):
+                t = max(bezier_params_from_point(point, control_points))
```

Comment:
i can skew towards long names but instead of `t` i'd call it `proportion_along_nth_curve`

Reply:
I've made it call `proportions_along_bezier_curve_for_point` once and use that value instead of calling both `point_lies_on_bezier` and `proportions_along_bezier_curve_for_point`